src/sortStudents.ts

```diff
@@ -1,16 +1,72 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'dsc';
 
+export function sortStudents(students: Student[], sortBy: SortType, order: SortOrder): Student[] {
+  const arr: Student[] = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        return arr.sort((a, b) => a.name.toLowerCase().localeCompare(b.surname.toLowerCase()));
```

Comment:
use sortBy for the field name
then you can combine case SortType.Name and case SortType.Surname, cause they will be identical (sort strings)

```suggestion
        return arr.sort((a, b) => a[sortBy].toLowerCase().localeCompare(b[sortBy].toLowerCase()));
```